lib/hex/update_checker.ex

```diff
@@ -9,19 +9,19 @@ defmodule Hex.UpdateChecker do
 
   def start_link(opts \\ []) do
     opts = Keyword.put_new(opts, :name, @name)
-    GenServer.start_link(__MODULE__, [], opts)
+    GenServer.start_link(__MODULE__, opts, opts)
```

Comment:
I think this is fine as is but maybe worth to differentiate between our internal options and the genserver ones? 

```suggestion
    {init_opts, opts} = Keyword.pop(opts, :init_opts, [])
    GenServer.start_link(__MODULE__, init_opts, opts)
```

wdyt @ericmj? if sounds good I'll fix it up and merge.

Reply:
So the advantage is we’d not pass init option like :init_state to the GenServer which it does not understand. Similarly we’d not pass a GenServer option like :name to the init which it doesn’t understand either.


Btw if this sounds good to you and you’d like to make this change, of course feel free @avillen!

Reply:
IMHO opinion, this makes a lot of sense. This will avoid sending data to code that won't use and understand. I can make the change these days!

Reply:
I agree, the options should be separated between internal and GenServer.

Reply:
@wojtekmach I have added a commit with this. Makes sense for you? Do not hesitate to request any other change